docs/DeveloperGuide.adoc

```diff
@@ -239,85 +238,147 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
+
+* **Alternative 1 (current choice):** Saves only the names of the Persons in the Attendees of Events.
+** Pros: Saves space. Able to filter EventList easily.
+** Cons: Requires going through the entire list of AddressBook to obtain details of persons attending the particular event.
+
+* **Alternative 2:** Saves the entire information of Events for each Persons.
+** Pros: Fast access to information of Events attended by any Persons.
+** Cons: Takes up a lot of storage space and memory. Many repeated items within the storage files.
+
+* **Alternative 3:** Assign an ID to each Persons and store in Attendees of Events.
+** Pros: Able to edit names in a straightforward coding manner.
+** Cons: We must add or assign an ID during creation of Person. This may require some refactoring of the current code base. Unnecessary if name is unique and uneditable.
+
+===== Aspect: Data structure to support `invite` command
+
+* **Alternative 1 (current choice):** Use `HashSet` to store persons names.
+** Pros: Easier and faster to retrieve or validate names stored. Able to handle duplicates easily.
+** Cons: Sequence of persons added is lost. Currently, this feature is not important for the project.
+
+* **Alternative 2:** Use `ArrayList` to store persons names.
+** Pros: Names are stored in order of addition.
+** Cons: Inefficient when handling data.
+
+==== Future Improvements
+
+===== Adding multiple persons to an event
 
-Step 5. The user then decides to execute the command `list`. Commands that do not modify the address book, such as `list`, will usually not call `Model#commitAddressBook()`, `Model#undoAddressBook()` or `Model#redoAddressBook()`. Thus, the `addressBookStateList` remains unchanged.
+The current `invite` command only invites one person to a chosen event based on the index chosen. This feature can be further extended to enable inviting more than one person to an event.
+One possible way to to implement this is to have the user
 
-image::UndoRedoNewCommand3StateListDiagram.png[width="800"]
+* input a certain range (2-5) or
+* multiple indices (1, 3, 6) or
+* a mixture of both (1-3, 4, 6-10)
 
-Step 6. The user executes `clear`, which calls `Model#commitAddressBook()`. Since the `currentStatePointer` is not pointing at the end of the `addressBookStateList`, all address book states after the `currentStatePointer` will be purged. We designed it this way because it no longer makes sense to redo the `add n/David ...` command. This is the behavior that most modern desktop applications follow.
+The same concept can also be applied to events, being able to invite one person to multiple events with one single command.
 
-image::UndoRedoNewCommand4StateListDiagram.png[width="800"]
+Implementation of this additional feature would require changes to both `InviteCommandParser` and `InviteCommand#execute`.
+// end::Invite[]
 
-The following activity diagram summarizes what happens when a user executes a new command:
+// tag:: Sorting[]
+
+=== Event Sorting
```

Comment:
Are section 3.1 and 3.3 referring to the same thing? As both are using the same header Event Sorting.